python/pyspark/java_gateway.py

```diff
@@ -51,13 +51,16 @@ def launch_gateway():
         on_windows = platform.system() == "Windows"
         script = "./bin/spark-submit.cmd" if on_windows else "./bin/spark-submit"
         submit_args = os.environ.get("PYSPARK_SUBMIT_ARGS", "pyspark-shell")
+        if conf and conf.getAll():
+            submit_args = ' '.join(['--conf %s="%s"' % (k, v) for k, v in conf.getAll()]) \
```

Comment:
This seems like it could be rather brittle - what happens if a user has a " inside one of the configuration values?